src/v/config/configuration.cc

```diff
@@ -1051,13 +1051,25 @@ configuration::configuration()
   , storage_compaction_key_map_memory(
       *this,
       "storage_compaction_key_map_memory",
-      "Maximum number of bytes that may be used on each shard by compaction"
-      "key-offset maps",
-      {.needs_restart = needs_restart::no,
+      "Maximum number of bytes that may be used on each shard by compaction "
+      "key-offset maps. Only respected when "
+      "`log_compaction_use_sliding_window` is true.",
+      {.needs_restart = needs_restart::yes,
        .example = "1073741824",
        .visibility = visibility::tunable},
       128_MiB,
       {.min = 16_MiB, .max = 100_GiB})
+  , storage_compaction_key_map_memory_limit_percent(
+      *this,
+      "storage_compaction_key_map_memory_limit_percent",
+      "Maximum percentage of memory per shard that may be used on each shard "
+      "by compaction key-offset maps. Only respected when "
+      "`log_compaction_use_sliding_window` is true.",
+      {.needs_restart = needs_restart::yes,
+       .example = "12.0",
+       .visibility = visibility::tunable},
+      12.0,
+      {.min = 1.0, .max = 100.0})
```

Comment:
Probably useful to mention how these properties relate to each other, as they both mention they are "Maximum"s.